lib/curlx/fopen.c

```diff
@@ -181,15 +183,17 @@ static bool fix_excessive_path(const TCHAR *in, TCHAR **out)
 
 #ifndef _UNICODE
   /* convert unicode full path to multibyte output */
-  needed = wcstombs(NULL, fbuf, 0);
-  if(needed == (size_t)-1 || needed >= max_path_len)
+  if(wcstombs_s(&needed, NULL, 0, fbuf, 0))
+    goto cleanup;
+  if(needed >= max_path_len)
     goto cleanup;
   ++needed; /* for NUL */
   obuf = (malloc)(needed);
   if(!obuf)
     goto cleanup;
-  count = wcstombs(obuf, fbuf, needed);
-  if(count == (size_t)-1 || count >= needed)
+  if(wcstombs_s(&count, obuf, needed, fbuf, _TRUNCATE))
+    goto cleanup;
+  if(count >= needed)
     goto cleanup;
```

Comment:
```suggestion
  if(count != needed)
    goto cleanup;
```
According to the [doc](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/wcstombs-s-wcstombs-s-l?view=msvc-170) the `count` written by the function cannot be greater than the passed in count + NUL (aka `needed - 1` + 1 for nul or in otherwords `needed`). I think that if the function succeeds. count should always equal needed unless I'm misreading